passes/changedlines/changedlines.go

```diff
@@ -0,0 +1,272 @@
+package changedlines
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+const servicePathPrefix = "internal/services/"
+
+var (
+	useGitRepo = flag.Bool("use-git-repo", true, "use git repository to calculate diff")
+	remoteName = flag.String("remote", "", "remote name (default: auto-detect)")
+	baseBranch = flag.String("base-branch", "", "base branch (default: main)")
+	diffFile   = flag.String("diff-file", "", "path to a diff file to parse")
+
+	useGitHubAPI = flag.Bool("use-github-api", false, "use GitHub API to get PR changes")
+	prNumber     = flag.Int("pr-number", 0, "GitHub PR number")
+	repoName     = flag.String("repo-name", "terraform-provider-azurerm", "GitHub repository name")
+
+	mu           sync.RWMutex
+	changedLines map[string]map[int]bool
+	changedFiles map[string]bool
+	newFiles     map[string]bool
+	initialized  bool
+
+	hunkRegex = regexp.MustCompile(`^@@ -\d+(?:,\d+)? \+(\d+)(?:,(\d+))? @@`)
+)
+
+// Initialize sets up the changed lines tracking system
+func Initialize() error {
+	mu.Lock()
+	defer mu.Unlock()
+
+	changedLines = make(map[string]map[int]bool)
+	changedFiles = make(map[string]bool)
+	newFiles = make(map[string]bool)
+
+	// Check if user provided a diff file
+	if diffFile != nil && *diffFile != "" {
+		if err := initializeFromDiffFile(*diffFile); err != nil {
+			return fmt.Errorf("failed to initialize from diff file: %w", err)
+		}
+	} else if useGitRepo != nil && *useGitRepo {
+		if err := initializeFromGitRepoSmart(); err != nil {
+			return fmt.Errorf("failed to initialize from git repo: %w", err)
+		}
+	}
+
+	initialized = true
+	return nil
+}
+
+// initializeFromGitRepoSmart chooses the best git-based initialization method
+func initializeFromGitRepoSmart() error {
+	if isGitHubActions() && canUseGitHubAPI() {
+		fmt.Fprintf(os.Stderr, "Detected GitHub Actions with PR context\n")
+		return initializeFromGitHubAPI()
+	}
+
+	if useGitHubAPI != nil && *useGitHubAPI {
```

Comment:
`useGitHubAPI` will always non-nil while defiend by `flag.Bool`
```suggestion
	if *useGitHubAPI {
```